Framework/Algorithms/test/PDFFourierTransform2Test.h

```diff
@@ -260,6 +289,132 @@ class PDFFourierTransform2Test : public CxxTest::TestSuite {
+    // set the sample information
+    Mantid::DataHandling::SetSample setsample;
+    setsample.initialize();
+    setsample.setProperty("InputWorkspace", ws);
+    setsample.setProperty("Material", material);
+    setsample.execute();
+
+    // check g(r) returns correct value
+    DataObjects::Workspace2D_sptr pdfws_gr = run_pdfft2_alg(ws, "g(r)", "Forward");
+    const auto little_gofR = pdfws_gr->y(0);
+    const auto gofR_comparison = 3.2015617108;
+    TS_ASSERT_DELTA(little_gofR[10], gofR_comparison, 1e-8);
+
+    // check G(r) returns correct value
+    DataObjects::Workspace2D_sptr pdfws_big_gr = run_pdfft2_alg(ws, "G(r)", "Forward");
+    const auto big_gofR = pdfws_big_gr->y(0);
+    const auto R_G = pdfws_big_gr->x(0);
+    const auto calculated_big_gofR = (gofR_comparison - 1) * 4. * M_PI * rho0 * R_G[10];
+    TS_ASSERT_DELTA(big_gofR[10], calculated_big_gofR, 1e-8);
+
+    // check RDF(r) returns correct value
+    DataObjects::Workspace2D_sptr pdfws_rdf_r = run_pdfft2_alg(ws, "RDF(r)", "Forward");
+    const auto rdfofR = pdfws_rdf_r->y(0);
+    const auto R_RDF = pdfws_rdf_r->x(0);
+    const auto calculated_rdfofR = gofR_comparison * 4. * M_PI * rho0 * R_RDF[10] * R_RDF[10];
+
+    TS_ASSERT_DELTA(rdfofR[10], calculated_rdfofR, 1e-8);
+
+    //// check G_k(r) returns correct value
+    DataObjects::Workspace2D_sptr pdfws_gkr = run_pdfft2_alg(ws, "G_k(r)", "Forward");
+    const auto gkofR = pdfws_gkr->y(0);
+    const Kernel::Material &material2 = pdfws_gkr->sample().getMaterial();
+    const auto factor = 0.01 * pow(material2.cohScatterLength(), 2);
+    const auto calculated_gkofR = (gofR_comparison - 1) * factor;
+
+    TS_ASSERT_DELTA(gkofR[10], calculated_gkofR, 1e-8);
+  }
+
+  void test_PDFTypes_bkwd() {
+    API::MatrixWorkspace_sptr ws = createWS(20, 0.1, "CheckResult3", "AtomicDistance");
+
+    // shared values for tests
+    const double single_x = 2.0;
+    std::vector<double> x(2, single_x);
+    std::vector<double> dy(2, 0.0);
+    std::vector<double> dx(2, 0.0);
+    const double rho0 = 1.0;
+    const double cohScatLen = 1.0;
+    const double factor1 = 4. * M_PI * rho0;
+
+    // set up initial y values
+    std::vector<double> y(2, 5.0);
+
+    // Algorithm destructor crashes without workspace properties initialised
+    PDFFourierTransform2 pdfft;
+    pdfft.initialize();
+    pdfft.setProperty("InputWorkspace", ws);
+    pdfft.setProperty("OutputWorkspace", "outputWS");
+
+    // test g(r)
+    const std::string LITTLE_G_OF_R("g(r)");
+    pdfft.convertToLittleGRMinus1(y, x, dy, dx, LITTLE_G_OF_R, rho0, cohScatLen);
+    const auto exp_gr = 4.0;
+    const auto actual_gr = y[0];
+    TS_ASSERT_DELTA(actual_gr, exp_gr, 1e-8);
+
+    // test G(r) - note that y values have been changed by previous call to convertToLittleGRMinus1
+    const std::string BIG_G_OF_R("G(r)");
+    pdfft.convertToLittleGRMinus1(y, x, dy, dx, BIG_G_OF_R, rho0, cohScatLen);
+    const auto exp_big_gr = exp_gr / (factor1 * single_x);
```

Comment:
Couple of things relating to these backward tests:

1) please can you rename this `exp_big_gr` variable. It actually holds a quantity little g(r). So strictly speaking it is `exp_little_gr_that_has_been_obtained_from_big_g_r_input` but since that's a bit of a mouthful perhaps simply calling it `exp_little_gr` is fine?

2) Please can you have a go at not chaining together the tests? The fact that the y values output from the previous test get used as the input for the next one is a bit confusing. 

If you declare your initial vector higher up on L357 as:
```
std::vector<double> y_initial(2, 5.0);
```
...and then take a copy of it before each test it should work. So for the following test do something like:
```
auto y = y_initial;
pdfft.convertToLittleGRMinus1(y, x, dy, dx, ...);
const auto exp_big_gr = y_initial / (factor1 * single_x);
```
so I don't think this will inolve changing all the formulae that you have for your exp_xxx variables